### ISSUE TITLE
Keep DNS rewrites active with "disabled protection"

### ISSUE THREAD

**Issue by coffinbearer** [AdguardTeam/AdGuardHome#1558]
### Problem Description
I would like to have the option to have my dns rewrites active and "disable the ad protection".

### Proposed Solution
add a option to keep dns rewrites active while disabeling ad protection.

### Alternatives Considered
change the default way dns rewites work and alwyas keep them active.

### Additional Information
Version of AdGuard Home server:
    v0.101 (Docker tag :latest)

**Comment by ArchiveBase** [comment]
Go to 'General Settings' and uncheck 'Block domains using filters and host files'
If you want to disable certain DNS rewrite temporally then you can suggest about On Off button for for each rewrite row or something else however you need to edit this issue or open separate feature request for that.

**Comment by lbalogh** [comment]
Why has this request been closed ?

I access AdGuard web interface by going to **https://adguard.mydomain.com**. **adguard.mydomain.com** is defined in the "DNS rewrites" section.

If I disable protection for some reason (like debugging a problematic website), **adguard.mydomain.com** doesn't resolve anymore and I lose access to the web interface (and all other services that are defined as DNS rewrites)

**Comment by ameshkov** [comment]
@lbalogh please upvote this issue if you want this to work that way.

**Comment by lbalogh** [comment]
Done. Thanks !

**Comment by Raytestrak** [comment]
I'd like this too, because all my apps on my NAS make use of url rewrites and I need to disable adguard sometimes (when using some smart tv apps, which don't anything in the query log, but will only work when adguard is suspended). Funny thing is, when I disable adguard, my dns rewrite is unavailable (say: adguard.dnssuffix) but immediately after I access adguard via my public domain (addguard.mydomainwhichistoolongtotypehencetheurlrewrites.com), the rewrite works again?

**Comment by Archive5** [comment]
@everyone
In general settings page has option to disable 'block domains using filters and hosts files'

**Comment by Raytestrak** [comment]
> @everyone
> In general settings page has option to disable 'block domains using filters and hosts files'

I know how to disable / pause adguard and I don't think that is the OP's question as well. I do it automatically via a webhook when I launch the troublesome apps on my smart tv, my 'problem' is that all of my webapps on my nas are unavailabe via the dns rewrite after turning off adguard due to fact the dns rewrites stop working when disabling adguard.

**Comment by Archive5** [comment]
'Block domains using filters and hosts files' is the solution for @coffinbearer @lbalogh 
AGH needs to have feature for disable all components then "disable protection" for it

@Raytestrak You have two problems, @ameshkov may take care of it
>I need to disable adguard sometimes (when using some smart tv apps, which don't anything in the query log, but will only work when adguard is suspended).

>when I disable adguard, my dns rewrite is unavailable (say: adguard.dnssuffix) but immediately after I access adguard via my public domain (addguard.urlrewrites.com), the rewrite works again?

Optionally, it's good to add feature for enable or disable each DNS rewrites same as in blocklist page

**Comment by lbalogh** [comment]
@Archive5 You're right, in my case 'block domains using filters and hosts files' is the solution.

But I still think that the "Disable protection" button on the main page should disable, as it says, **protection**. DNS rewrites is not "protection" in my opinion.

Maybe the label could be changed to "Disable AdGuard Home", OR the option to completely disable AdGuard Home (dns rewrites included) should be moved to an other place

**Comment by Raytestrak** [comment]
@Archive5 disabling block lists probably could be the solution to my 1st problem, but like @lbalogh said, DNS rewrites isn't part of protection. The top four checkboxes under general settings are, though.  

Right now I use **curl -X POST -H "Content-Type:application/json" -d '{"protection_enabled":false}' http://192.168.1.252:9080/control/dns_config** to apparently disable everyting. 

I'll change it to **curl -X POST -H "Content-Type:application/json" -d '{"enabled":false}' http://192.168.1.252:9080/control/filtering/config** and see what happens. That probably won't interfere with my dns rewrites (my 2nd problem). But ... I still think this feature should still be active when disabling protection.

**Comment by SilentGob** [comment]
Got bitten by this and it took me too long to understand what was happening so I will just agree with the orignal poster. 
Rewrite is (in my configuration) not part of protection and I would prefer that disabling protection for testing purpose does not shutdown rewrite.

**Comment by zilexa** [comment]
This is a horrible bug: everyone runs AGH on their home network and can use DNS rewrite to easily access.. the web UI of AGH!!

And probably all their other services running on their network.

After I disable protection, I can't even access AdGuardHome anymore! 

Please fix. Note the bug is worse: after enabling protection, the DNS rewrite still doesn't work. I have to restart AGH!

**Comment by zilexa** [comment]
Can this please be solved in the next version? 
Everytime I disable protection, I cannot enable it anymore because it was disabled via the pretty url (dns-rewrite) to adguard on my server.

**Comment by Dameon87** [comment]
Would definitely like to see this implemented too. DNS rewrites should not be disabled when protection is disabled.

**Comment by ainar-g** [comment]
This will probably be done together with #2499.

**Comment by lbalogh** [comment]
Good news, thanks for the follow up !

**Comment by EugeneOne1** [comment]
@coffinbearer @lbalogh @Raytestrak @SilentGob @zilexa @Dameon87
Here it is.  The latest build in the edge channel improves the filtering logic.  `$dnsrewrite` rules and other DNS rewrites are applied to each request even with the protection being disabled.  Those now are controlled only by the **Block domains using filters and hosts files** checkbox in **General settings**.

Could you please install the build and check, if our solution serves your purposes properly?  Thanks.

**Comment by chaelli** [comment]
Hi @EugeneOne1 

should this be in every version starting from 0.107? I just tested on v0.107.11 (from docker) and the behavior seems to be the same as with the old v0.106.x version - rewrites are disabled if I click the "disable protection" button.

**Comment by felkal** [comment]
Hi there,

This is still a problem for me running Adguard Home v0.107.43. When I disable protection for a particular device, it can no longer reach services that are accessible due to DNS rewrites.

**Comment by rubin110** [comment]
I just created #7803, which is related to this behavior but when protection is disabled per client.
